### PR TITLE
Fix bug in ignoring new version available update banner

### PR DESCRIPTION
**Pull request by ephraimbuddy**
Tested this manually as I was unable to do unittest on it

### CODE REVIEW

**Review by ashb — COMMENTED** [review]
I think we only allowed admins to dismiss the notice. Is that still kept?

**Comment by jedcunningham** [comment]
Any idea how the admin role got can_dismiss though? Having trouble tracing that back.

**Comment by ashb** [comment]
Oh, admin gets all permissions aged by FAB at start up

**Comment by ephraimbuddy** [comment]
> Any idea how the admin role got `can_dismiss` though? Having trouble tracing that back.

From FAB, there's a [PERMISSION_PREFIX='can_'](https://github.com/dpgaspar/Flask-AppBuilder/blob/c7477aa4b0fa2744e73c4b730312e43b17697760/flask_appbuilder/const.py#L132) and then the method name of the endpoint 'dismiss'

**Comment by ephraimbuddy** [comment]
> I think we only allowed admins to dismiss the notice. Is that still kept?

Nope. Let me get one of the admins permissions and replace

File: astronomer/airflow/version_check/update_checks.py [hunk 1]
```diff
@@ -381,9 +382,19 @@ class UpdateAvailable(BaseApi):
...
+        method_permission_name = {
+            "dismiss": "dismiss",
+        }
+        allow_browser_login = True
 
         @expose("<path:version>/dismiss", methods=["POST"])
-        @protect(allow_browser_login=True)
+        @auth.has_access(
+            [
+                (permissions.ACTION_CAN_READ, permissions.RESOURCE_WEBSITE),
```
Comment (ephraimbuddy) [inline astronomer/airflow/version_check/update_checks.py:395]:
```suggestion
                (permissions.ACTION_CAN_READ, permissions.RESOURCE_ROLE),
```

File: astronomer/airflow/version_check/update_checks.py [hunk 2]
```diff
@@ -381,9 +382,19 @@ class UpdateAvailable(BaseApi):
...
+        method_permission_name = {
+            "dismiss": "dismiss",
+        }
+        allow_browser_login = True
 
         @expose("<path:version>/dismiss", methods=["POST"])
-        @protect(allow_browser_login=True)
+        @auth.has_access(
+            [
+                (permissions.ACTION_CAN_READ, permissions.RESOURCE_ROLE),
```
Comment (jedcunningham) [inline astronomer/airflow/version_check/update_checks.py:395]: Can we use whatever is under `can dismiss on UpdateAvailable` instead?

Comment (ephraimbuddy) [reply astronomer/airflow/version_check/update_checks.py:395]: Not sure how to get it

Comment (ephraimbuddy) [reply astronomer/airflow/version_check/update_checks.py:395]: Will check tomorrow

Comment (ephraimbuddy) [reply astronomer/airflow/version_check/update_checks.py:395]: It also seems like it won’t be possible except we create the permission and add it to Admin

Comment (ephraimbuddy) [reply astronomer/airflow/version_check/update_checks.py]: Turns out that ('can_dismiss', 'update_available') works.

Comment (ephraimbuddy) [reply astronomer/airflow/version_check/update_checks.py]: This works `('can_dismiss', 'UpdateAvailable')`

**Review by ashb — APPROVED** [review]

**Pull request merged**